code/modules/station_goals/dna_vault.dm

```diff
@@ -0,0 +1,190 @@
+	item_state = "hypo"
+	icon_state = "hypo"
+	flags = NOBLUDGEON
+	var/list/animals = list()
+	var/list/plants = list()
+	var/list/dna = list()
+
+/obj/item/device/dna_probe/proc/clear_data()
+	animals = list()
+	plants = list()
+	dna = list()
+
+var/list/non_simple_animals = typecacheof(list(/mob/living/carbon/monkey)|subtypesof(/mob/living/carbon/alien))
+
+/obj/item/device/dna_probe/afterattack(atom/target, mob/user, proximity)
+	..()
+	if(!proximity || !target)
+		return
+	//tray plants
+	if(istype(target,/obj/machinery/hydroponics))
+		var/obj/machinery/hydroponics/H = target
+		if(!H.myseed)
+			return
+		if(!H.harvest)// So it's bit harder.
+			user << "<span clas='warning'>Plants needs to be ready to harvest to perform full data scan.</span>" //Because space dna is actually magic
+			return
+		if(plants[H.myseed.type])
+			user << "<span class='notice'>Plant data already present in local storage.<span>"
+			return
+		plants[H.myseed.type] = 1
+		user << "<span class='notice'>Plant data added to local storage.<span>"
+
+	//animals
+	if(isanimal(target) || is_type_in_typecache(target.type,non_simple_animals))
+		if(isanimal(target))
+			var/mob/living/simple_animal/A = target
+			if(!A.healable)//simple approximation of being animal not a robot or similar
+				user << "<span class='warning'>No compatibile DNA detected</span>"
+				return
+		if(animals[target.type])
+			user << "<span class='notice'>Animal data already present in local storage.<span>"
+			return
+		animals[target.type] = 1
+		user << "<span class='notice'>Animal data added to local storage.<span>"
+
+	//humans
+	if(ishuman(target))
+		var/mob/living/carbon/human/H = target
+		if(dna[H.dna.uni_identity])
+			user << "<span class='notice'>Humanoid data already present in local storage.<span>" 
+			return
+		dna[H.dna.uni_identity] = 1
+		user << "<span class='notice'>Humanoid data added to local storage.<span>"
+
+
+/obj/item/weapon/circuitboard/machine/dna_vault
+	name = "circuit board (DNA Vault)"
+	build_path = /obj/machinery/dna_vault
+	origin_tech = "engineering=2;combat=2;bluespace=2" //No freebies!
+	req_components = list(
+							/obj/item/weapon/stock_parts/capacitor/quadratic = 5,
+							/obj/item/stack/cable_coil = 2)
+
+/obj/machinery/dna_vault
+	name = "DNA Vault"
+	desc = "Break glass in case of apocalypse."
+	icon = 'icons/obj/machines/dna_vault.dmi'
+	icon_state = "vault"
+	density = 1
+	anchored = 1
+	idle_power_usage = 5000
+	pixel_x = -32
+	pixel_y = -64
+	luminosity = 1
+	var/list/animals = list()
+	var/list/plants = list()
+	var/list/dna = list()
+
+	var/list/obj/structure/fillers = list()
+
+/obj/machinery/dna_vault/New()
+	//TODO: Replace this,bsa and gravgen with some big machinery datum
+	var/list/occupied = list()
+	for(var/direct in list(EAST,WEST,SOUTHEAST,SOUTHWEST))
+		occupied += get_step(src,direct)
+	occupied += locate(x+1,y-2,z)
+	occupied += locate(x-1,y-2,z)
+
+	for(var/T in occupied)
+		var/obj/structure/filler/F = new(T)
```

Comment:
We don't use bounds because if you get it even 1 value wrong it all explodes and dies.
but if you get it perfectly multiples of ```world.icon_size``` (32 for us) it works fine.

plus there's talk from lummox/byond about making pixel/turf movement a FIXED world setting, which should avoid similar issues in the future.